ark/type/src/parser/tuple.ts

```diff
@@ -133,39 +141,44 @@ export type validateTuple<
 	  ]
 	: validateTupleLiteral<def, $, args>
 
+type writeIllegalTupleElement<def extends string> =
```

Comment:
We should just get rid of this custom error for now. There's several changes that would need to be made to have it work consistently between string and tuple optional suffixes, so probably not worth it.